### PR TITLE
[MM-55305] Migrate "server/channels/store/sqlstore/post_store.go" to use GenericStoreResult

### PR DESCRIPTION
**Pull request by Paul-Stern**
#### Summary
Make use of GenericStoreResult in "server/channels/store/sqlstore/post_store.go".

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25308
Jira https://mattermost.atlassian.net/browse/MM-55305

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/store/sqlstore/post_store.go [hunk 1]
```diff
@@ -2753,7 +2753,7 @@ func (s *SqlPostStore) SearchPostsForUser(paramsList []*model.SearchParams, user
 		if result.NErr != nil {
 			return nil, result.NErr
 		}
-		data := result.Data.(*model.PostList)
+		data := result.Data
 		posts.Extend(data)
```
Comment (hanzei) [inline server/channels/store/sqlstore/post_store.go:2757]: Nit: We can simplify the code here
```suggestion
		posts.Extend(result.Data)
```

**Review by hanzei — APPROVED** [review]
Another one!

Thanks for the work, @Paul-Stern :+1:

**Review by agnivade — APPROVED** [review]
Good to go after Ben's suggestion is taken care of.

**Pull request merged**